### PR TITLE
[API-1372]: Create scene with scene items

### PR DESCRIPTION
**Pull request by therockstorm**
## Summary

Add an `--experimental` flag to `create-scene` that creates with `scene-items` instead of from a template.

## Possible Regressions

n/a

## Dependencies

n/a

### CODE REVIEW

File: src/commands/create-scene.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
 import { flags } from '@oclif/command';
```
Comment (therockstorm) [inline src/commands/create-scene.ts:1]: Ignore whitespace

File: tsconfig.json [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
     "importHelpers": true,
+    "lib": ["es2020"],
     "module": "commonjs",
+    "noImplicitAny": true,
     "outDir": "lib",
     "rootDir": "src",
     "strict": true,
-    "target": "es2020",
-    // Remove `dom` once there's an acceptable fix for https://github.com/axios/axios/issues/3219
-    "lib": ["dom", "es2020"]
```
Comment (therockstorm) [inline tsconfig.json:12]: The issue is fixed, we can remove `dom`

File: src/create-template/pvs.ts [hunk 3]
```diff
@@ -156,22 +158,24 @@ function createItems(
...
   const t = args.transform;
 
   return {
+    depth: args.pathId.split(PathIdSeparator).length - 1,
     fileName: args.fileName,
-    name: args.partName,
-    parentId: parentId === '' ? '/' : parentId,
-    source: `/parts?filter[suppliedId]=${encodeURI(
+    suppliedPartId: args.partName,
+    suppliedRevisionId: args.partRevision,
```
Comment (therockstorm) [inline src/create-template/pvs.ts:172]: Pull these into separate fields so I don't have to decode the `source` to get them when creating parts.

File: src/create-template/pvs.ts [hunk 4]
```diff
@@ -156,22 +158,24 @@ function createItems(
...
+    parentId: parentId === '' ? PathIdSeparator : parentId,
+    source: `/parts?filter[suppliedId]=${encodeURIComponent(
       args.partName
-    )}&include=part-revisions&filter[part-revisions][suppliedId]=${
+    )}&include=part-revisions&filter[part-revisions][suppliedId]=${encodeURIComponent(
```
Comment (danschultz) [inline src/create-template/pvs.ts:176]: Consider using Node's [`URLSearchParams`](https://nodejs.org/api/url.html#url_class_urlsearchparams) here. Also includes url encoding.

File: tsconfig.json [hunk 5]
```diff
@@ -3,13 +3,14 @@
     "declaration": true,
     "esModuleInterop": true,
     "importHelpers": true,
+    "lib": ["es2020"],
```
Comment (danschultz) [inline tsconfig.json:6]: Consider sticking with ES2019. v12 only supports ES2019 and is still an active LTS.

File: tsconfig.json [hunk 6]
```diff
@@ -3,13 +3,14 @@
...
     "module": "commonjs",
+    "noImplicitAny": true,
     "outDir": "lib",
     "rootDir": "src",
     "strict": true,
-    "target": "es2020",
-    // Remove `dom` once there's an acceptable fix for https://github.com/axios/axios/issues/3219
-    "lib": ["dom", "es2020"]
+    "strictNullChecks": true,
+    "target": "es2020"
```
Comment (danschultz) [inline tsconfig.json:13]: ^

**Review by danschultz — APPROVED** [review]
Some considerations. Otherwise LGTM.

File: tsconfig.json (hunk 5 above)
Comment (therockstorm) [reply tsconfig.json:6]: I need ES2020 for `Promise.allSettled`

File: tsconfig.json (hunk 6 above)
Comment (therockstorm) [reply tsconfig.json:13]: Can I use `es2020` for the `lib` and target `es2019`?

File: src/create-template/pvs.ts (hunk 4 above)
Comment (therockstorm) [reply src/create-template/pvs.ts]: Updated

File: tsconfig.json (hunk 6 above)
Comment (therockstorm) [reply tsconfig.json]: Updated to `es2019`

**Review by danschultz — APPROVED** [review]

**Pull request merged**